src/ai/backend/manager/api/auth.py

```diff
@@ -1129,6 +1130,10 @@ async def upload_ssh_keypair(request: web.Request, params: Any) -> web.Response:
     log_args = (domain_name, access_key)
     log.info(log_fmt, *log_args)
     root_ctx: RootContext = request.app["_root.context"]
+
+    is_valid, err_msg = validate_ssh_keypair(privkey, pubkey)
+    if not is_valid:
+        raise InvalidAPIParameters(err_msg)
```

Comment:
Wouldn't it be better to throw an exception internally?

Reply:
Raising `InvalidAPIParameters` turns into returning 400 HTTP response because the `InvalidAPIParameters` inherits HTTPBadRequest exception

Reply:
Well... we'll clean up the exception in a later pick.